### PR TITLE
Add support for Codeberg as a git resolver

### PR DESCRIPTION
**Pull request by miry**
Enhances the `shard.yml` configuration to support [Codeberg](https://codeberg.org/),
a popular Git hosting service in Germany, as a dependency source.

This allows developers to specify Codeberg repositories directly in their configuration.

Example usage:

```yaml
dependencies:
  foo:
    codeberg: repo/foo
```

### CODE REVIEW

File: src/resolvers/git.cr [hunk 1]
```diff
@@ -100,7 +100,15 @@ module Shards
...
-    private KNOWN_PROVIDERS = {"www.github.com", "github.com", "www.bitbucket.com", "bitbucket.com", "www.gitlab.com", "gitlab.com"}
+    private KNOWN_PROVIDERS = {
+      "www.github.com",
+      "github.com",
+      "www.bitbucket.com",
+      "bitbucket.com",
+      "www.gitlab.com",
+      "gitlab.com",
+      "codeberg.org",
+    }
```
Comment (devnote-dev) [inline src/resolvers/git.cr:111]: I'd suggest including `www.codeberg.org` too for support.

```suggestion
      "www.codeberg.org",
      "codeberg.org",
    }
```

**Review by devnote-dev — CHANGES_REQUESTED** [review]

**Review by straight-shoota — APPROVED** [review]

File: man/shard.yml.5 [hunk 2]
```diff
@@ -524,7 +524,8 @@ A path to the source file to compile (string).
 .SH "DEPENDENCY ATTRIBUTES"
 .sp
 Each dependency needs at least one attribute that defines the resolver for this
-dependency. Those can be \fIpath\fP, \fIgit\fP, \fIgithub\fP, \fIgitlab\fP, \fIbitbucket\fP.
+dependency. Those can be \fIpath\fP, \fIgit\fP, \fIgithub\fP, \fIgitlab\fP, \fIbitbucket\fP,
+\fIcodeberg\fP.
```
Comment (Sija) [inline man/shard.yml.5:528]:
```suggestion
dependency. Those can be \fIpath\fP, \fIgit\fP, \fIgithub\fP, \fIgitlab\fP, \fIbitbucket\fP, \fIcodeberg\fP.
```

Comment (straight-shoota) [reply man/shard.yml.5:528]: The man pages should be generated with `make manpages` and contain whatever the generator produces.

**Review by devnote-dev — APPROVED** [review]

File: man/shard.yml.5 (hunk 2 above)
Comment (miry) [reply man/shard.yml.5:528]: My bad, I am going to fix in 2 hours.

Comment (miry) [reply man/shard.yml.5]: Updated the man pages.

**Review by straight-shoota — APPROVED** [review]

**Review by ysbaddaden — APPROVED** [review]

**Pull request merged**